### PR TITLE
Calling `shipit` in PR in CI creates canary release

### PR DESCRIPTION
**Pull request by hipstersmoothie**
# What Changed

see title

# Why

task in #346

Todo:

- [ ] Add tests
- [ ] Add docs
- [ ] Add yourself to contributors (run `yarn contributors:add`)

### CODE REVIEW

**Comment by hipstersmoothie** [comment]
@adierkens @zephraph Do you guys think this would be a breaking change?

Previously you could run `auto shipit` from any branch and it *might* work depending on the git tree state. 

Now `auto` will only publish to the latest tag when we are on master.

**Comment by adierkens** [comment]
Published PR with canary version: `4.14.0-canary.351.4292`

**Comment by adierkens** [comment]
Published PR with canary version: `4.14.0-canary.351.4297`

**Comment by adierkens** [comment]
Published PR with canary version: `4.14.0-canary.351.4302`

**Comment by zephraph** [comment]
I'd say it is, yeah. It might be good to give folks an option to configure their head branch. Otherwise, release as a major.

File: src/auto.ts [hunk 1]
```diff
@@ -492,9 +495,39 @@ export default class Auto {
     this.logger.verbose.info("Using command: 'shipit'");
     this.hooks.beforeShipIt.call();
 
+    const isPR = 'isPr' in env && env.isPr;
+    // env-ci sets branch to target branch (ex: master) in some CI services.
+    // so we should make sure we aren't in a PR just to be safe
+    const isMaster = !isPR && 'branch' in env && env.branch === 'master';
```
Comment (zephraph) [inline src/auto.ts]: Probably wouldn't be too hard to go ahead and make this `isBaseBranch` and set a `BASE_BRANCH` const somewhere to compare it to. We can make it configurable in a separate PR.

File: src/auto.ts [hunk 2]
```diff
@@ -492,9 +495,39 @@ export default class Auto {
...
+    const publishInfo = isPR
+      ? await this.canary(options)
+      : isMaster
+      ? await this.publishLatest(options)
+      : undefined;
+
+    if (!isPR && !isMaster) {
+      this.logger.log.info(
+        'No version published. Could not detect if in PR or on master branch.'
```
Comment (zephraph) [inline src/auto.ts]: Just update the wording here to be `base branch` or use the `BASE_BRANCH` const mentioned above.

**Comment by adierkens** [comment]
Published PR with canary version: `5.0.0-canary.351.4346`

File: src/auto.ts (hunk 1 above)
Comment (hipstersmoothie) [reply src/auto.ts]: I have made the base branch configurable in this PR https://github.com/intuit/auto/pull/357.

File: src/auto.ts (hunk 2 above)
Comment (hipstersmoothie) [reply src/auto.ts]: https://github.com/intuit/auto/pull/357 constructs the string with whatever the base branch is. Do you think it would be better to just say base branch?

**Comment by adierkens** [comment]
Published PR with canary version: `5.0.0-canary.351.4434`

**Comment by adierkens** [comment]
:rocket: PR was released in v5.0.0 :rocket:
